Docker/StorageApi/Reader.php

```diff
@@ -74,18 +78,28 @@ public function __construct(Client $client)
      */
     public function downloadFiles($configuration, $destination)
     {
-        $options = new ListFilesOptions();
-        if (isset($configuration["tags"]) && count($configuration["tags"])) {
-            $options->setTags($configuration["tags"]);
-        }
-        if (isset($configuration["query"])) {
-            $options->setQuery($configuration["query"]);
-        }
-        $files = $this->getClient()->listFiles($options);
-        foreach ($files as $file) {
-            $fileInfo = $this->getClient()->getFile($file["id"], (new GetFileOptions())->setFederationToken(true));
-            $this->downloadFile($fileInfo, $destination . "/" . $fileInfo["id"]);
-            $this->writeFileManifest($fileInfo, $destination . "/" . $fileInfo["id"] . ".manifest");
+        foreach ($configuration as $fileConfiguration) {
+            $options = new ListFilesOptions();
+            if (empty($fileConfiguration['tags']) && empty($fileConfiguration['query'])) {
+                throw new UserException("Invalid file mapping, both 'tags' and 'query' are empty.");
+            }
+            if (isset($fileConfiguration["tags"]) && count($fileConfiguration["tags"])) {
+                $options->setTags($fileConfiguration["tags"]);
+            }
+            if (isset($fileConfiguration["query"])) {
+                $options->setQuery($fileConfiguration["query"]);
+            }
+            $files = $this->getClient()->listFiles($options);
+
+            // a little sanity check, otherwise it may easily happen that a wrong ES query would fill up the server
```

Comment:
zajimavej point. 10 souborů mi možná přijde i málo, ale nechal bych to tak, uvidíme, jestli to bude na někoho řvát a pak se to kdyžtak fixne.